src/main/java/cuchaz/enigma/translation/representation/entry/ClassDefEntry.java

```diff
@@ -28,11 +28,16 @@
 	private final ClassEntry[] interfaces;
 
 	public ClassDefEntry(String className, Signature signature, AccessFlags access, @Nullable ClassEntry superClass, ClassEntry[] interfaces) {
-		this(getOuterClass(className), getInnerName(className), signature, access, superClass, interfaces);
+		this(getOuterClass(className), getInnerName(className), signature, access, superClass, interfaces, null);
 	}
 
 	public ClassDefEntry(ClassEntry parent, String className, Signature signature, AccessFlags access, @Nullable ClassEntry superClass, ClassEntry[] interfaces) {
-		super(parent, className);
+		this(parent, className, signature, access, superClass, interfaces, null);
+	}
+
+	public ClassDefEntry(ClassEntry parent, String className, Signature signature, AccessFlags access, @Nullable ClassEntry superClass,
+						 ClassEntry[] interfaces, String javadocs) {
```

Comment:
As something optional, maybe `javadocs` should rather be set by a setter, instead of coming through the constructor?

Reply:
Disregard this, see below